### PR TITLE
[TT-7573] add pluginConfing.driver

### PR DESCRIPTION
**Pull request by jeffy-mathew**
## Description
Add pluginConfig.driver

## Related Issue
https://tyktech.atlassian.net/browse/TT-7573

### CODE REVIEW

File: apidef/oas/middleware.go [hunk 1]
```diff
@@ -38,6 +38,9 @@ func (m *Middleware) ExtractTo(api *apidef.APIDefinition) {
 
 // Global holds configuration applies globally: CORS and caching.
 type Global struct {
+	// PluginConfig contains the configuration related custom plugin bundles/driver.
+	PluginConfig *PluginConfig `bson:"pluginConfig" json:"pluginConfig"`
```
Comment (furkansenharputlu) [inline apidef/oas/middleware.go:42]: `,omitempty`?

**Comment by Tyk-ITS** [comment]
**API tests result: success** :white_check_mark: 
Branch used: refs/pull/4656/merge
Commit:  
Triggered by: pull_request (@jeffy-mathew)
[Execution page](https://github.com/TykTechnologies/tyk/actions/runs/3937922911)

**Comment by Tyk-ITS** [comment]
**API tests result: success** :white_check_mark: 
Branch used: refs/pull/4656/merge
Commit: 18332e643e456be221dc315faaad232288f8c40f 
Triggered by: pull_request (@jeffy-mathew)
[Execution page](https://github.com/TykTechnologies/tyk/actions/runs/3938201948)

File: apidef/oas/middleware.go (hunk 1 above)
Comment (jeffy-mathew) [reply apidef/oas/middleware.go:42]: missed it, thanks :)

**Review by furkansenharputlu — APPROVED** [review]

**Comment by Tyk-ITS** [comment]
**API tests result: success** :white_check_mark: 
Branch used: refs/pull/4656/merge
Commit: 31d3f4d7f2dcc569bdab2692cc7de54dbf80fbe0 
Triggered by: pull_request (@jeffy-mathew)
[Execution page](https://github.com/TykTechnologies/tyk/actions/runs/3938478347)

File: apidef/oas/schema/x-tyk-gateway.md [hunk 2]
```diff
@@ -660,6 +660,10 @@ Operations configuration.
 
 ### **Global**
 
+**Field: `pluginConfig` ([PluginConfig](#pluginconfig))**
+
+PluginConfig contains the configuration related custom plugin bundles/driver.
```
Comment (titpetric) [inline apidef/oas/schema/x-tyk-gateway.md:665]: Typo, missing "to" before custom

File: apidef/oas/schema/x-tyk-gateway.md [hunk 3]
```diff
@@ -673,6 +677,23 @@ Cache contains the configurations related to caching.
 Tyk native API definition: `cache_options`.
 
 
+### **PluginConfig**
+
+**Field: `driver` (`object`)**
+
+Driver configures which custom plugin to be used.
```
Comment (titpetric) [inline apidef/oas/schema/x-tyk-gateway.md:684]: *to use.

File: apidef/oas/middleware.go [hunk 4]
```diff
@@ -38,6 +38,9 @@ func (m *Middleware) ExtractTo(api *apidef.APIDefinition) {
 
 // Global holds configuration applies globally: CORS and caching.
 type Global struct {
+	// PluginConfig contains the configuration related custom plugin bundles/driver.
```
Comment (titpetric) [inline apidef/oas/middleware.go:41]: "related to"

**Review by titpetric — APPROVED** [review]

**Pull request merged**